akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1583,6 +1583,9 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
     override def toString = s"SubSourceOutlet($name)"
   }
 
+  override def toString: String =
+    attributes.get[Attributes.Name].map(attr => s"${getClass.toString}-${attr.n}").getOrElse(getClass.toString)
+
```

Comment:
Can we move it before the inner classes to the other methods?

Reply:
Sure, How about [here](https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala#L1006)?

Reply:
You know what, this class is so incomprehensibly big anyway, just leave it where it is.

Reply:
I just did :)